akka-docs/src/main/paradox/stream/operators/Source-or-Flow/foldAsync.md

```diff
@@ -13,7 +13,23 @@ Just like `fold` but receives a function that results in a @scala[`Future`] @jav
 
 Just like `fold` but receives a function that results in a @scala[`Future`] @java[`CompletionStage`] to the next value.
 
-Note that the `zero` value must be immutable.
+@@@ warning
+
+Note that the `zero` value must be immutable, because otherwise
+the same mutable instance would be shared across different threads
+when running the stream more than once.
+
+@@@
+
+## Example
+
+`fold` is typically used to 'fold up' the incoming values into an aggregate asynchronously. For example, you might want to summarize the incoming values into a histogram:
```

Comment:
fold -> foldAsync

If we keep the async histogram as folding operation I think this should point out that it is not an actual way you'd use futures/completion stages but that it is for the sake of the example.